### PR TITLE
Z_obj_bell Matching

### PR DESCRIPTION
**Pull request by jpburnett**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

Looking forward to everyone chiming in on this one. I tried to run graphovl for the cool picture, but it said "No actor action-based structure found"...

👹 👉 🔔

### CODE REVIEW

File: src/overlays/actors/ovl_Obj_Bell/z_obj_bell.c [hunk 1]
```diff
@@ -21,26 +29,259 @@ const ActorInit Obj_Bell_InitVars = {
...
+                if (temp_f0 > 0.18f) {
+                    func_800B8D98(globalCtx, &this->dyna.actor, 8.0f * temp_f0, this->dyna.actor.yawTowardsPlayer,
+                                  11.0f * temp_f0);
+                    this->unk_214 = 30;
+                }
+            }
+        }
+    }
+    if ((this->collider2.base.acFlags & 2)) {
+        this->collider2.base.acFlags &= 0xFFFD;
```
Comment (engineer124) [inline src/overlays/actors/ovl_Obj_Bell/z_obj_bell.c:169]:
```suggestion
    if (this->collider2.base.acFlags & 2) {
        this->collider2.base.acFlags &= ~2;
```

File: src/overlays/actors/ovl_Obj_Bell/z_obj_bell.c [hunk 2]
```diff
@@ -21,26 +29,259 @@ const ActorInit Obj_Bell_InitVars = {
...
+    }
+    return 0;
+}
+
+s32 func_80A357A8(ObjBell* this, GlobalContext* globalCtx) {
+    f32 temp_f0;
+    s16 temp_v1;
+
+    if ((this->collider1.base.ocFlags1 & 2)) {
+        this->collider1.base.ocFlags1 &= 0xFFFD;
```
Comment (engineer124) [inline src/overlays/actors/ovl_Obj_Bell/z_obj_bell.c:155]:
```suggestion
    if (this->collider1.base.ocFlags1 & 2) {
        this->collider1.base.ocFlags1 &= ~2;
```

File: src/overlays/actors/ovl_Obj_Bell/z_obj_bell.c [hunk 3]
```diff
@@ -21,26 +29,259 @@ const ActorInit Obj_Bell_InitVars = {
...
+    s16 phi_v1;
+
+    this->collider1.dim.worldSphere.radius = (this->collider1.dim.modelSphere.radius * this->collider1.dim.scale);
+    this->collider2.dim.worldSphere.radius = (this->collider2.dim.modelSphere.radius * this->collider2.dim.scale);
+
+    // TODO: This temp could be taken out, but do we want it in for readability?...
+    phi_v1 = DECR(this->unk_20E);
+    if (phi_v1 == 0) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colCheckCtx, &this->collider2.base);
+    }
```
Comment (engineer124) [inline src/overlays/actors/ovl_Obj_Bell/z_obj_bell.c:198]:
```suggestion
    if (DECR(this->unk_20E) == 0) {
        CollisionCheck_SetAC(globalCtx, &globalCtx->colCheckCtx, &this->collider2.base);
    }
```
Yeah, I'd take this temp out:
https://discord.com/channels/688807550715560050/688851997352263820/844406587870937119

File: src/overlays/actors/ovl_Obj_Bell/z_obj_bell.c [hunk 4]
```diff
@@ -21,26 +29,259 @@ const ActorInit Obj_Bell_InitVars = {
...
+}
+
+void func_80A359B4(Actor* thisx, GlobalContext* globalCtx) {
+    SysMatrix_InsertTranslation(thisx->world.pos.x, thisx->world.pos.y, thisx->world.pos.z, 0);
+    Matrix_Scale(thisx->scale.x, thisx->scale.y, thisx->scale.z, 1U);
+    SysMatrix_InsertTranslation(0.0f, 2600.0f, 0.0f, 1);
+    Matrix_RotateY(thisx->world.rot.y, 1U);
+    SysMatrix_InsertXRotation_s(thisx->world.rot.x, 1);
+    Matrix_RotateY(-thisx->world.rot.y, 1U);
+    SysMatrix_InsertTranslation(0.0f, -2600.0f, 0.0f, 1);
```
Comment (engineer124) [inline src/overlays/actors/ovl_Obj_Bell/z_obj_bell.c:209]:
```suggestion
    SysMatrix_InsertTranslation(thisx->world.pos.x, thisx->world.pos.y, thisx->world.pos.z, MTXMODE_NEW);
    Matrix_Scale(thisx->scale.x, thisx->scale.y, thisx->scale.z, MTXMODE_APPLY);
    SysMatrix_InsertTranslation(0.0f, 2600.0f, 0.0f, MTXMODE_APPLY);
    Matrix_RotateY(thisx->world.rot.y, MTXMODE_APPLY);
    SysMatrix_InsertXRotation_s(thisx->world.rot.x, MTXMODE_APPLY);
    Matrix_RotateY(-thisx->world.rot.y, MTXMODE_APPLY);
    SysMatrix_InsertTranslation(0.0f, -2600.0f, 0.0f, MTXMODE_APPLY);
```

File: src/overlays/actors/ovl_Obj_Bell/z_obj_bell.c [hunk 5]
```diff
@@ -21,26 +29,259 @@ const ActorInit Obj_Bell_InitVars = {
...
+    gSPDisplayList(POLY_OPA_DISP++, D_060008D0);
+    gSPDisplayList(POLY_OPA_DISP++, D_06000960);
+    gSPDisplayList(POLY_OPA_DISP++, D_060007A8);
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void func_80A35B18(Actor* thisx, GlobalContext* globalCtx) {
+    SysMatrix_InsertTranslation(thisx->world.pos.x, thisx->world.pos.y, thisx->world.pos.z, NULL);
+    Matrix_Scale(thisx->scale.x, thisx->scale.y, thisx->scale.z, 1);
+    Matrix_RotateY(thisx->shape.rot.y, 1);
```
Comment (engineer124) [inline src/overlays/actors/ovl_Obj_Bell/z_obj_bell.c:224]:
```suggestion
    SysMatrix_InsertTranslation(thisx->world.pos.x, thisx->world.pos.y, thisx->world.pos.z, MTXMODE_NEW);
    Matrix_Scale(thisx->scale.x, thisx->scale.y, thisx->scale.z, MTXMODE_APPLY);
    Matrix_RotateY(thisx->shape.rot.y, MTXMODE_APPLY);
```

File: src/overlays/actors/ovl_Obj_Bell/z_obj_bell.c [hunk 6]
```diff
@@ -21,26 +29,259 @@ const ActorInit Obj_Bell_InitVars = {
...
+    func_8012C28C(globalCtx->state.gfxCtx);
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_OPA_DISP++, D_06000570);
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Bell_0x80A35510/func_80A357A8.asm")
+void func_80A35BD4(Actor* thisx, GlobalContext* globalCtx) {
+    SysMatrix_InsertTranslation(thisx->world.pos.x, thisx->world.pos.y - 4.0f, thisx->world.pos.z, 0);
+    Matrix_Scale(thisx->scale.x, thisx->scale.y, thisx->scale.z, 1);
```
Comment (engineer124) [inline src/overlays/actors/ovl_Obj_Bell/z_obj_bell.c:235]:
```suggestion
    SysMatrix_InsertTranslation(thisx->world.pos.x, thisx->world.pos.y - 4.0f, thisx->world.pos.z, MTXMODE_NEW);
    Matrix_Scale(thisx->scale.x, thisx->scale.y, thisx->scale.z, MTXMODE_APPLY);
```

**Review by engineer124 — COMMENTED** [review]
Great work! All Stock Pot Inn guests can now enjoy the shinny new bell on the balcony (well except for the Goron who's room key was so tragically stolen). Just a few minor suggestions below.

File: src/overlays/actors/ovl_Obj_Bell/z_obj_bell.c [hunk 7]
```diff
@@ -21,26 +29,259 @@ const ActorInit Obj_Bell_InitVars = {
...
+
+    this->unk_212 = this->dyna.actor.world.rot.y;
+    if (this->unk_20C >= 0x4000 || this->unk_20C < -0x3FFF) {
+        this->unk_212 -= 0x8000;
+    }
+    temp_f2 = Math_SinS(this->unk_20C) * this->unk_21C;
+    this->dyna.actor.world.rot.x = this->dyna.actor.home.rot.x;
+    new_var = temp_f2;
+    this->unk_220 = temp_f2;
+    this->dyna.actor.world.rot.x += new_var;
```
[the remark this answers is not in the document]
Comment (jpburnett) [reply src/overlays/actors/ovl_Obj_Bell/z_obj_bell.c:142]: it does!

File: src/overlays/actors/ovl_Obj_Bell/z_obj_bell.c [hunk 8]
```diff
@@ -21,26 +29,259 @@ const ActorInit Obj_Bell_InitVars = {
     (ActorFunc)ObjBell_Update,
     (ActorFunc)ObjBell_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Bell_0x80A35510/func_80A35510.asm")
+// Bell Post?
+ColliderSphereInit sCylinderInit1 = {
```
[the remark this answers is not in the document]
Comment (jpburnett) [reply src/overlays/actors/ovl_Obj_Bell/z_obj_bell.c:34]: thanks for catching that, thought I put them in 👍

File: src/overlays/actors/ovl_Obj_Bell/z_obj_bell.c [hunk 9]
```diff
@@ -21,26 +29,250 @@ const ActorInit Obj_Bell_InitVars = {
...
+            case 14:
+                Audio_PlayActorSound2(this, 0x28F3);
+                func_80A35510(this, 2);
+                break;
+            default:
+                func_80A35510(this, 0);
+                break;
+        }
+    }
+    return false;
```
Comment (engineer124) [inline src/overlays/actors/ovl_Obj_Bell/z_obj_bell.c:183]: Thanks for the fixes :) one last thought. Three functions, `func_80A35510`, `func_80A356D8`, and `func_80A357A8`, all have return values that seem unnecessary, as the return is never used when any of these functions are called. Sometimes a return is required to match, but is there any chance you can remove the `return false` switch to `void func_` for any of them and still get a match?

Comment (jpburnett) [reply src/overlays/actors/ovl_Obj_Bell/z_obj_bell.c:183]: image

**Review by engineer124 — APPROVED** [review]

**Review by Zelllll — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**